app/assets/javascripts/components/widget-wrapper.js

```diff
@@ -0,0 +1,85 @@
+ManageIQ.angular.app.component('widgetWrapper', {
+  bindings: {
+    widgetId: '@',
+    widgetType: '@',
+    widgetButtons: '@',
+    widgetBlank: '@',
+    widgetTitle: '@',
+    widgetLastRun: '@',
+    widgetNextRun: '@',
+  },
+  controllerAs: 'vm',
+  controller: ['$http', 'miqService', '$sce', function($http, miqService, $sce) {
+    var vm = this;
+
+    var deferred = miqDeferred();
+    vm.promise = deferred.promise;
+
+    this.$onInit = function() {
+      vm.divId = "w_" + vm.widgetId;
+      vm.innerDivId = 'dd_w' + vm.widgetId + '_box';
+      if (vm.widgetBlank === 'false') {
+        $http.get(vm.widgetUrl())
+          .then(function(response) {
+            vm.widgetModel = response.data;
+            // if there's html make it passable
+            if (vm.widgetModel.content) {
+              vm.widgetModel.content =  $sce.trustAsHtml(vm.widgetModel.content);
+            }
+            deferred.resolve();
+          })
+          .catch(function(e) {
+            vm.error = true;
+            miqService.handleFailure(e);
+            deferred.reject();
+          });
+      }
+    };
+
+    vm.widgetUrl  = function() {
+      switch (vm.widgetType) {
+        case 'menu':
+          return '/dashboard/widget_menu_data/' + vm.widgetId;
```

Comment:
It's preferred to have a whole url to make it easier to find it. That's why it's not pretty.

Reply:
Maybe it still makes sense to use an object instead of a switch. But yes, please keep the whole url together, this will be a part of the component definition later.